### DISCUSSION TITLE
Ego graph with actual data?

### DISCUSSION THREAD

**Discussion by ajmeese7** [networkx/networkx#7601] [Q&A] [answered]
I am looking at the [documentation](https://networkx.org/documentation/stable/auto_examples/drawing/plot_ego_graph.html) and I see that it is being used on a Barabási-Albert network with randomly-generated data, but can I use this layout on non-randomized data that I provide?

Sorry if it's a silly question, I am still getting familiar with the NetworkX ecosystem.

**Comment by rossbar** [comment, marked as answer]
Yes, `ego_graph` can be used to create the ego_graph for any node in any networkx graph. The [documentation for `ego_graph` itself](https://networkx.org/documentation/stable/reference/generated/networkx.generators.ego.ego_graph.html) should (hopefully) make that more clear than the referenced gallery example.

A quick example with "real" data, e.g. the [one of the social network datasets from the SNAP database](https://snap.stanford.edu/data/ego-Facebook.html):

```python
>>> import networkx as nx
>>> with open("3980.edges") as fh:  # arbitrary dataset from the database
...     edgelist = [line.rstrip().split(" ") for line in fh.readlines()]
>>> G = nx.Graph(edgelist)
>>> ego_4037 = nx.ego_graph(G, "4037")
>>> ego_4037.edges
EdgeView([('3982', '4014'), ('3982', '4037'), ('4020', '4037'), ('4037', '4014')])
```

**Reply by ajmeese7** [reply]
I will look into this now, thank you so much for your help and quick response!
